### PR TITLE
Init Project

### PR DESCRIPTION
**Pull request by umairah97**

### CODE REVIEW

**Comment by umairah97** [comment]
ready

**Comment by umairah97** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
+source 'https://rubygems.org'
```
Comment (chungpv-1008) [inline Gemfile:1]: remove all comments in this file

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
+
+  <body>
+    <%= yield %>
+  </body>
+</html>
```
Comment (chungpv-1008) [inline app/views/layouts/application.html.erb:15]: use double quotes instead of single quote in rb, erb files

File: config/application.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+module SampleApp
+  class Application < Rails::Application
+    # Initialize configuration defaults for originally generated Rails version.
+    config.load_defaults 5.2
+
+    # Settings in config/environments/* take precedence over those specified here.
+    # Application configuration can go into files in config/initializers
+    # -- all .rb files in that directory are automatically loaded after loading
+    # the framework and any gems in your application.
```
Comment (chungpv-1008) [inline config/application.rb:17]: replace " with '
remove all comments in this file

File: config/credentials.yml.enc [hunk 4]
```diff
@@ -0,0 +1 @@
+ELwoAGSNDTiBR6wtiQp5ziavk6ygdBu1gx5YNLDjwIxgBoRWN38C8Tg02HOxgtEDFzSzBGzMcYAyci0r787Ex2RSPHseKdT50P/i4tSzG4/ACyWLF9qUugnLislSiwTcsf5ukdBAz3FstPiYLcQeutRfPrHJAGg7PjOYk5pZFN8jesNewqHas22sYIhh4egQ8vqsG0t4IIctKsZVlJq71ChikgqeJ7j2XtDA/K4k3Vn87kmiIVSIxUX+O2zvwdyGONJLcdW4z30aCaS97wZ7kRqn88gQtl2WJrzau7VRS3Y6oGwxeOp9QdFLvPc+Gw+bd8h16HLrB8LIelKU3yurV/QNYnt5KrjxDNwAJKsW6UW9UX1Xuu2QSustC3bj2Koj4erozNb/NiSGgxBYX/Pkv9r7rTcwO3sIdvhr--G3JfNGDiZe/LzYoB--ZcHEnU6LuZ9f3zEeLfWGng==
```
Comment (chungpv-1008) [inline config/credentials.yml.enc:1]: You move this file to the '.gitignore' file. Create 1 empty file 'credentials.yml.enc.example'

File: config/database.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+#   gem install sqlite3
+#
+#   Ensure the SQLite 3 gem is defined in your Gemfile
+#   gem 'sqlite3'
+#
+default: &default
+  adapter: sqlite3
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  timeout: 5000
+
```
Comment (chungpv-1008) [inline config/database.yml:11]: remove all comments this file. You use 'mysql2' gem to instead of 'sqlite3'

File: config/routes.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
```
Comment (chungpv-1008) [inline config/routes.rb:2]:
```suggestion
```

File: test/application_system_test_case.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require "test_helper"
```
Comment (chungpv-1008) [inline test/application_system_test_case.rb:1]: Ignore folder 'test' in file '.gitignore'. you don't need to follow it anymore

File: Gemfile (hunk 1 above)
Comment (umairah97) [reply Gemfile:1]: done

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (umairah97) [reply app/views/layouts/application.html.erb]: done

File: config/application.rb (hunk 3 above)
Comment (umairah97) [reply config/application.rb]: done

File: config/database.yml (hunk 5 above)
Comment (umairah97) [reply config/database.yml:11]: done

File: config/credentials.yml.enc (hunk 4 above)
Comment (umairah97) [reply config/credentials.yml.enc:1]: done

File: test/application_system_test_case.rb (hunk 7 above)
Comment (umairah97) [reply test/application_system_test_case.rb:1]: done

**Comment by umairah97** [comment]
ready

**Comment by umairah97** [comment]
ready

File: config/credentials.yml.enc (hunk 4 above)
Comment (umairah97) [reply config/credentials.yml.enc]: done

File: .gitignore [hunk 8]
```diff
@@ -54,3 +54,9 @@ build-iPhoneSimulator/
 
 # Used by RuboCop. Remote config files pulled in from inherit_from directive.
 # .rubocop-https?--*
+
+# Ignore master key for decrypting credentials and more.
+/config/master.key
+credentials.yml.enc.example
+database.yml
+test
```
[the remark this answers is not in the document]
Comment (umairah97) [reply .gitignore]: done

File: .ruby-version [hunk 9]
```diff
@@ -0,0 +1 @@
+ruby-2.7.0
```
[the remark this answers is not in the document]
Comment (umairah97) [reply .ruby-version]: done

File: app/mailers/application_mailer.rb [hunk 10]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: 'from@example.com'
+  layout 'mailer'
```
[the remark this answers is not in the document]
Comment (umairah97) [reply app/mailers/application_mailer.rb]: done

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>SampleApp</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track": "reload" %>
```
[the remark this answers is not in the document]
Comment (umairah97) [reply app/views/layouts/application.html.erb]: done

**Comment by umairah97** [comment]
ready

File: .gitignore [hunk 12]
```diff
@@ -54,3 +54,10 @@ build-iPhoneSimulator/
 
 # Used by RuboCop. Remote config files pulled in from inherit_from directive.
 # .rubocop-https?--*
+
+# Ignore master key for decrypting credentials and more.
+/config/master.key
+config/credentials.yml.enc.example
+config/database.yml
+test
+end
```
Comment (saobv-2223) [inline .gitignore:63]: image

File: .ruby-version [hunk 13]
```diff
@@ -0,0 +1,2 @@
...
+end
```
Comment (saobv-2223) [inline .ruby-version:2]: delete `end`

File: config/locales/en.yml [hunk 14]
```diff
@@ -0,0 +1,33 @@
+# Files in the config/locales directory are used for internationalization
+# and are automatically loaded by Rails. If you want to use locales other
+# than English, add the necessary files in this directory.
+#
+# To use the locales, use `I18n.t`:
```
Comment (saobv-2223) [inline config/locales/en.yml:5]: remove all the comment lines

**Review by saobv-2223 — COMMENTED** [review]
Run command to merge all the commits to ONE commit
`git reset --soft origin/master`
`git commit -m <commit>`
`git push origin -f init_project`

File: .gitignore (hunk 12 above)
Comment (umairah97) [reply .gitignore]: done

**Comment by umairah97** [comment]
ready

**Review by chungpv-1008 — COMMENTED** [review]
how do many files change like this ?
Why is there an folder 'git'

**Comment by umairah97** [comment]
ready

**Review by chungpv-1008 — COMMENTED** [review]
approved

**Comment by chungpv-1008** [comment]
merge

**Comment by vanvtt-0952** [comment]
merged

**Pull request merged**